statsd/statsd_test.go

```diff
@@ -198,6 +198,136 @@ func TestBufferedClient(t *testing.T) {
 
 }
 
+func TestJoinMaxSize(t *testing.T) {
```

Comment:
I think I fixed a few things in the test in the other refactor PR after you started your work. Would you mind bringing those in? Unfortunately I squashed those, so it might be a bit of a mess. Alternatively I could rebase to your changes there....